### PR TITLE
feat(backfill): start via CLI arguments

### PR DESCRIPTION
**Pull request by shekhirin**
image

### CODE REVIEW

File: backfill/src/main.rs [hunk 1]
```diff
@@ -223,20 +208,61 @@ impl<Node: FullNodeComponents> BackfillExEx<Node> {
...
+            process_committed_chain(&chain).await
+        })
+        .await
+}
 
-        info!(first_block = %chain.execution_outcome().first_block, %blocks, %transactions, "Processed committed chain");
-        Ok(())
-    }
+/// Processes the committed chain and logs the number of blocks and transactions.
+async fn process_committed_chain(chain: &Chain) -> eyre::Result<()> {
```
Comment (mattsse) [inline backfill/src/main.rs:244]: this doesn need async?

File: backfill/src/main.rs [hunk 2]
```diff
@@ -252,8 +278,45 @@ fn main() -> eyre::Result<()> {
...
+                // Rust seems to trigger a bogus higher-ranked lifetime error when using
+                // just an async closure here -- using `spawn_blocking` avoids this
+                // particular issue.
+                //
+                // To avoid the higher ranked lifetime error we use `spawn_blocking`
+                // which will move the closure to another blocking-allowed thread,
+                // then execute.
+                //
+                // Source: https://github.com/vados-cosmonic/wasmCloud/commit/440e8c377f6b02f45eacb02692e4d2fabd53a0ec
+                tokio::task::spawn_blocking(move || {
```
Comment (mattsse) [inline backfill/src/main.rs:291]: I believe this equivalent to ctx.task_executor().spawn_blocking().await

**Review by mattsse — APPROVED** [review]
cool, 
one suggestion but not sure if using the task executor also works

File: backfill/src/main.rs (hunk 2 above)
Comment (shekhirin) [reply backfill/src/main.rs:291]: yeah, but task executor methods accept futures that resolve to `()`, but we need the return value here

**Pull request merged**